<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ur5e_robot.urdf.xacro          | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur5e">
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/base.dae"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.163"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/shoulder.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.04"/>
      <geometry>
        <cylinder length="0.12" radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.019999999999999997"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.1"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.5707963267948966 0.0" xyz="0.0 0.138 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/upperarm.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 -0.04" xyz="0 -0.04 0"/>
      <geometry>
        <cylinder length="0.12" radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.019999999999999997 0"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 -0.1 0"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <cylinder length="0.4" radius="0.05"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.4"/>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.0"/>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.2125"/>
      <inertia ixx="0.1338857818623325" ixy="0.0" ixz="0.0" iyy="0.1338857818623325" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.131 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/forearm.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0.08" xyz="0 0.08 0"/>
      <geometry>
        <cylinder length="0.12" radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.14 0"/>
      <geometry>
        <sphere radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.020000000000000004 0"/>
      <geometry>
        <sphere radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <cylinder length="0.38" radius="0.038"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.39"/>
      <geometry>
        <sphere radius="0.038"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.010000000000000009"/>
      <geometry>
        <sphere radius="0.038"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.196"/>
      <inertia ixx="0.031179620861480004" ixy="0.0" ixz="0.0" iyy="0.031179620861480004" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.5707963267948966 0.0" xyz="0.0 0.0 0.392"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/wrist1.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0.05" xyz="0 0.05 0"/>
      <geometry>
        <cylinder length="0.14" radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.12000000000000001 0"/>
      <geometry>
        <sphere radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 -0.020000000000000004 0"/>
      <geometry>
        <sphere radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.127 0.0"/>
      <inertia ixx="0.0025598989760400002" ixy="0.0" ixz="0.0" iyy="0.0025598989760400002" iyz="0.0" izz="0.0021942"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.127 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/wrist2.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <cylinder length="0.12" radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 0.1"/>
      <geometry>
        <sphere radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 -0.019999999999999997"/>
      <geometry>
        <sphere radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="1.5707963267948966 0 0.02" xyz="0 0.02 0.1"/>
      <geometry>
        <cylinder length="0.08" radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0.06 0.1"/>
      <geometry>
        <sphere radius="0.04"/>
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 -0.02 0.1"/>
      <geometry>
        <sphere radius="0.04"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.1"/>
      <inertia ixx="0.0025598989760400002" ixy="0.0" ixz="0.0" iyy="0.0025598989760400002" iyz="0.0" izz="0.0021942"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_ros_sim/mujoco_menagerie/universal_robots_ur5e/assets/wrist3.dae"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="1.5707963267948966 0 0" xyz="0.0 0.0771 0.0"/>
      <inertia ixx="9.890410052167731e-05" ixy="0.0" ixz="0.0" iyy="9.890410052167731e-05" iyz="0.0" izz="0.0001321171875"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="0.0 0.0 1.5707963267948966" xyz="0.0 0.1 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
        <geometry>
          <box size="0.01 0.01 0.01"/>
        </geometry>
        <origin rpy="0 0 0" xyz="-0.01 0 0"/>
      </collision>
  </link>
</robot>
